charmcraft/application/commands/remote.py

```diff
@@ -0,0 +1,196 @@
+            "--launchpad-accept-public-upload",
+            action="store_true",
+            help="acknowledge that uploaded code will be publicly available.",
+        )
+        parser.add_argument(
+            "--launchpad-timeout",
+            type=int,
+            default=0,
+            metavar="<seconds>",
+            help="Time in seconds to wait for launchpad to build.",
+        )
+        parser.add_argument(
+            "--project-dir",
+            type=pathlib.Path,
+            default=None,
+            help="set the directory for the project",
+        )
+
+    def _run(self, parsed_args: argparse.Namespace, **kwargs: Any) -> int | None:  # noqa: ANN401
+        """Run the remote-build command.
+
+        :param parsed_args: parsed argument namespace from craft_cli.
+
+        :raises AcceptPublicUploadError: If the user does not agree to upload data.
+        """
+        if os.getenv("SUDO_USER") and os.geteuid() == 0:
+            emit.progress(
+                "Running with 'sudo' may cause permission errors and is discouraged.",
+                permanent=True,
+            )
+            time.sleep(1)
+
+        emit.progress(
+            "remote-build is experimental and is subject to change. Use with caution.",
+            permanent=True,
+        )
+
+        if not parsed_args.launchpad_accept_public_upload:
+            if not utils.confirm_with_user(_CONFIRMATION_PROMPT, default=False):
+                emit.message("Cannot proceed without accepting a public upload.")
+                return 77  # permission denied from sysexits.h
+
+        builder = self._services.remote_build
+        project = cast(models.Charm, self._services.project)
+        config = cast(dict[str, Any], self.config)
+        project_dir = (
+            pathlib.Path(config.get("global_args", {}).get("project_dir") or ".")
+            .expanduser()
+            .resolve()
+        )
+        emit.trace(f"Project directory: {project_dir}")
+
+        if parsed_args.launchpad_timeout:
+            emit.debug(f"Setting timeout to {parsed_args.launchpad_timeout} seconds")
+            builder.set_timeout(parsed_args.launchpad_timeout)
+
+        build_id = get_build_id(self._app.name, project.name, project_dir)
+        if parsed_args.recover:
+            emit.progress(f"Recovering build {build_id}")
+            builds = builder.resume_builds(build_id)
+        else:
+            emit.progress("Starting new build. It may take a while to upload large projects.")
+            builds = builder.start_builds(project_dir)
+
+        try:
+            returncode = self._monitor_and_complete(build_id, builds)
+        except KeyboardInterrupt:
+            if utils.confirm_with_user("Cancel builds?", default=True):
+                emit.progress("Cancelling builds.")
+                builder.cancel_builds()
+                emit.progress("Cleaning up")
+                builder.cleanup()
+            returncode = 0
```

Comment:
If user Ctrl+C multiple times, this going to skip. It shouldn't be a problem.

Reply:
Yep, that's intentional. If someone uses Ctrl+C multiple times they **really** want to quit and I don't want to get in their way. (Especially since the time between a keyboard interrupt and getting that confirmation prompt should be unnoticeable.)